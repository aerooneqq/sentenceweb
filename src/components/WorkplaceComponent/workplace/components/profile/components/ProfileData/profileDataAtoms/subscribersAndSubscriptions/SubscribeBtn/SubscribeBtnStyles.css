.subscribeBtn { 
    width: 80px;
    height: 30px;

    background: #12A949;

    border-style: solid;
    border-width: 1px;
    border-radius: 7px;
    border-color: #12A949;

    font-family: Arial;
    color: white;
}

    .subscribeBtn:focus{ 
        outline: none;
    }

    .subscribeBtn:hover{ 
        background: linear-gradient(-120deg, #28B95D, #12A949)
    }