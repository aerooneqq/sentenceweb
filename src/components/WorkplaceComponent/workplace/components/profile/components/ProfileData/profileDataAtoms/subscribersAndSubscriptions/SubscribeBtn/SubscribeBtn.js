import React, {Component} from "react";

//Styles
import "./SubscribeBtnStyles.css";

export default class SubscribeBtn extends Component { 
    constructor(props) { 
        super(props);

        this.handleBtnClick = this.handleBtnClick.bind(this);
    }

    handleBtnClick(){ 
        this.props.subscribe(this.props.userID);
    }

    render(){ 
        return ( 
            <button className = "subscribeBtn"
                    onClick = {this.handleBtnClick}>
                Subscribe
            </button>
        )
    }
}