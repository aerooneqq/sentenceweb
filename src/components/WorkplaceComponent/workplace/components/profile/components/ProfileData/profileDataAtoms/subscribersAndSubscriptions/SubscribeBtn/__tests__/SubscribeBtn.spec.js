import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import SubscribeBtn from "../SubscribeBtn";

const propsSpy = jest.fn((x) => {});
const props = { 
    userID: 1,
    subscribe: propsSpy
}

describe("SubscribeBtn", () => { 
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<SubscribeBtn />))).toMatchSnapshot();
    });

    it("should call subscribe from props when handleBtnClick is called", () => { 
        const subscribeBtn = shallow(<SubscribeBtn {...props} />);
        subscribeBtn.instance().handleBtnClick();


        expect(propsSpy).toBeCalled();
    });
    
    it("should call handleBtnClick when clicked", () => { 
        const spy = jest.spyOn(SubscribeBtn.prototype, "handleBtnClick").mockImplementation(() => {});
        const subscribeBtn = shallow(<SubscribeBtn />);

        subscribeBtn.find("button").simulate("click");

        expect(spy).toBeCalled();
    });
});