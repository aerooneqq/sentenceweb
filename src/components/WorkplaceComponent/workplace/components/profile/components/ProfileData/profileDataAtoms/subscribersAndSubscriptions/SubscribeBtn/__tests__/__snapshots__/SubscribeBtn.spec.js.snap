// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscribeBtn should render correctly 1`] = `
<button
  className="subscribeBtn"
  onClick={[Function]}
>
  Subscribe
</button>
`;
